### PR TITLE
Add clearer docs for how to build/sign application

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves https://github.com/microsoft/CCF/issues/554

### CODE REVIEW

File: cmake/common.cmake [hunk 1]
```diff
@@ -571,7 +571,7 @@ set(CCF_NETWORK_TEST_ARGS
   ${PBFT_ARG}
 )
 
-# Lua generic app
+# SNIPPET: Lua generic application
```
Comment (jumaffre) [inline cmake/common.cmake]: The logging example was a little confusing because of the `stub_for_code_signing.cpp` inclusion.

**Comment by cimetrics** [comment]
images

File: samples/apps/txregulator/app/oe_sign.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Enclave settings:
 Debug=1
-NumHeapPages=50000
+NumHeapPages=32768
```
[the remark this answers is not in the document]
Comment (achamayou) [reply samples/apps/txregulator/app/oe_sign.conf]: That's very bad, it didn't use to be the case. We need to report this and get it fixed asap.

**Comment by cimetrics** [comment]
images

File: samples/apps/txregulator/app/oe_sign.conf (hunk 2 above)
[the remark this answers is not in the document]
Comment (jumaffre) [reply samples/apps/txregulator/app/oe_sign.conf]: My bad. I was signing with the wrong config file.

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**